themes/bootstrap3/js/record.js

```diff
@@ -298,6 +298,74 @@ function removeCheckRouteParam() {
+  var target = $(_target);
+  var targetElemName = target.length && target.prop('tagName').toLowerCase();
+  var targetClass = target.length ? ` ${target.prop('class')}` : '';
+  var rowCount = holder.data('rows') || 3;
+  var moreLabel = holder.data('more-label') ? holder.data('more-label') : VuFind.translate('show_more');
+  var lessLabel = holder.data('less-label') ? holder.data('less-label') : VuFind.translate('show_less');
+  var btnSize = holder.data('btn-size') ? ` ${holder.data('btn-size')}` : '';
+  var topToggle = holder.data('top-toggle') || Infinity;
+  var inPlaceToggle = holder.data('in-place-toggle') || false;
+
+  holder.each(function parseHolder() {
+    var self = $(this);
+    var numRows = self.find(target).length || 0;
+    var hasTopToggle = numRows > topToggle;
+    // Truncate only if there's more than one line to hide
+    var shouldTruncate = rowCount < numRows || false;
+    self.find(target).each(function hideRows(i) {
+      var truncateAfter = rowCount - 1;
+      if (i == truncateAfter) {
+        $(this).addClass('truncate-after');
+      }
+      if(i > truncateAfter && shouldTruncate) {
+        $(this).hide();
+        $(this).addClass('truncate-toggle');
+      }
+    });
+
+    if (shouldTruncate) {
+      var btn = '<button type="button" class="more-link btn' + btnSize + '">' + moreLabel + ' <i class="fa fa-arrow-down" aria-hidden="true"></i></button><button type="button" class="less-link btn' + btnSize + '">' + lessLabel + ' <i class="fa fa-arrow-up" aria-hidden="true"></i></button>';
+      var btnLessTop = '<button type="button" class="less-link-top btn' + btnSize + '">' + lessLabel + ' <i class="fa fa-arrow-up" aria-hidden="true"></i></button>';
+      var btnWrapper = $('<' + targetElemName + ' class="more-less-btn-wrapper' + targetClass + '"></' + targetElemName + '>');
+      var btnWrapperBtm = btnWrapper.clone().append(_fill(btn));
+      var btnWrapperTop = btnWrapper.clone().append(_fill(btnLessTop));
+
+      // Attach show/hide buttons to the top and bottom or display in place
+      if (hasTopToggle) {
+        $(btnWrapperTop).prependTo(self);
+      }
+      if(inPlaceToggle) {
+        $(btnWrapperBtm).insertAfter(self.find('.truncate-after'));
+      } else {
+        $(btnWrapperBtm).appendTo(self);
+      }
+
+      self.find('.less-link, .less-link-top').hide();
+
+      self.find('.less-link, .less-link-top').click(function onClickLessLink(/*event*/) {
+        self.find('.less-link, .less-link-top').hide();
+        self.find('.more-link').show();
+        self.find('.truncate-toggle').toggle();
+      });
+
+      self.find('.more-link').click(function onClickMoreLink(/*event*/) {
+        $(this).hide();
+        self.find('.less-link').show();
+        if (hasTopToggle) {
+          self.find('.less-link-top').show();
+        }
+        self.find('.truncate-toggle').toggle();
+      });
+    }
+    self.addClass('truncate-done');
+    self.trigger('truncate-done', [self]);
```

Comment:
I might recommend using or at least also adding a `VuFind.emit('truncate-done', self);` to tie into the event system in the [VuFind core js](https://github.com/vufind-org/vufind/blob/fbf5028bfa261f5a653e5e54f114447d7d827d46/themes/bootstrap3/js/common.js#L14).

Reply:
I'm not sure if the event is actually needed. I don't think we're currently using it.

Reply:
Since it ties into my thoughts of the future theme, I would be in favor of extra descriptive events that make customization easier. I don't know if that applies to this particular event but the case could be made in general.